### PR TITLE
[IMP] account: changed the integer field to char field in fiscal posi…

### PR DESCRIPTION
**Pull request by rja-odoo**
Task: https://www.odoo.com/web#id=1948120&action=333&active_id=131&model=project.task&view_type=form&menu_id=4720
pad: https://pad.odoo.com/p/r.89d59b0033b0f57e50e3b9e53adc2082

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by qdp-odoo** [comment]
@rja-odoo so what? are you gonna fix your :hankey:  or should I do it myself?

File: addons/account/models/partner.py [hunk 1]
```diff
@@ -91,6 +85,38 @@ def _onchange_country_group_id(self):
...
+            if zip_from > zip_to:
+                raise ValidationError(_('Invalid "Zip Range", please configure it properly.'))
+            record.with_context({'zip_number_converted': True}).write({
+                'zip_from': zip_from,
+                'zip_to': zip_to
+            })
+        return
+
+    @api.model
+    def create(self, vals):
```
Comment (hmo-odoo) [inline addons/account/models/partner.py]: @rja-odoo , do not change the value of zip_from and zip_to on create/write method. we just need to validate the value of zip_from and zip_to in constraints.

so move stuff of _convert_zip_values into _check_zip method.

and @api.one is deprecated . so remove it from _check_zip method

Comment (hmo-odoo) [reply addons/account/models/partner.py]: @rja-odoo , wait

yes we need to change zip values on create/write method. 

In database, zip_from = "4" and zip_to = "10" if zipcode = "05" or zipcode = "5",  result will be False but it should be true

but In database, zip_from = "04" and zip_to = "10". zipcode = "05", result will be seem ok. but zipcode="5",  result will not ok.

@qdp-odoo , can we change zip values on create/write methods ? "4" => "04"

**Comment by rja-odoo** [comment]
> make _convert_zip_values() an @api.model method, with zip to convert as parameter. This way you can adapt values before the call to create() or write()



> make _convert_zip_values() an @api.model method, with zip to convert as parameter. This way you can adapt values before the call to create() or write()

Changes are done. Can you please have a look at it.

**Comment by qdp-odoo** [comment]
@robodoo r+

**Comment by qdp-odoo** [comment]
@robodoo retry

**Comment by qdp-odoo** [comment]
@robodoo r+

**Pull request closed without merging**